core/local/index.js

```diff
@@ -254,20 +272,31 @@ module.exports = class Local /*:: implements Side */ {
         // After downloading a file, check that the size is correct too
         // (more protection against stack corruption)
         if (!doc.size) return next()
-        fse.stat(tmpFile, (err, stats) => {
+        const cbStats = (err, stats) => {
           if (err) {
             next(err)
           } else if (doc.size === stats.size) {
+            doc.ino = stats.ino
+            if (stats.fileid) { doc.fileid = stats.fileid }
             next()
           } else {
             next(sentry.flag(new Error('Invalid size')))
           }
-        })
+        }
+        if (winfs) {
+          try {
+            const stats = winfs.lstatSync(tmpFile)
+            cbStats(null, stats)
+          } catch (err) {
+            cbStats(err, {})
+          }
+        } else {
+          fse.stat(tmpFile, cbStats)
+        }
       },
 
       next => fse.ensureDir(parent, () => fse.rename(tmpFile, filePath, next)),
 
-      this.inodeSetter(doc),
```

Comment:
When do we set the inode and fileId? Was this line useless?

Reply:
At lines 279 & 280. It's better to do it before moving the file from the tmp directory to its final destination: it's one less syscall to stats, and it avoids a possible race condition where the file is deleted between renaming and calling `inodeSetter`.

Reply:
It's now line 261